lib/pry/helpers/base_helpers.rb

```diff
@@ -1,138 +1,71 @@
+    $VERBOSE = nil
+    begin
+      yield
+    ensure
+      $VERBOSE = old_verbose
+    end
+  end
 
-      #
-      # @return [Boolean]
-      #   Returns true if Pry is running on Windows.
-      #
-      # @note
-      #   Queries RbConfig::CONFIG['host_os'] with a best guess.
-      #
-      def windows?
-        !!(RbConfig::CONFIG['host_os'] =~ /mswin|mingw/)
-      end
+  # Acts like send but ignores any methods defined below Object or Class in the
+  # inheritance hierarchy.
+  # This is required to introspect methods on objects like Net::HTTP::Get that
+  # have overridden the `method` method.
+  def safe_send(obj, method, *args, &block)
+    (Module === obj ? Module : Object).instance_method(method).bind(obj).call(*args, &block)
+  end
+  public :safe_send
 
-      # are we able to use ansi on windows?
-      def windows_ansi?
-        defined?(Win32::Console) || ENV['ANSICON'] || (windows? && mri_2?)
-      end
+  def find_command(name, set = Pry::Commands)
+    command_match = set.find do |_, command|
+      (listing = command.options[:listing]) == name && listing != nil
+    end
+    command_match.last if command_match
+  end
 
-      def jruby?
-        RbConfig::CONFIG['ruby_install_name'] == 'jruby'
-      end
+  def not_a_real_file?(file)
+    file =~ /^(\(.*\))$|^<.*>$/ || file =~ /__unknown__/ || file == "" || file == "-e"
+  end
 
-      def jruby_19?
-        jruby? && RbConfig::CONFIG['ruby_version'] == '1.9'
-      end
+  def command_dependencies_met?(options)
+    return true if !options[:requires_gem]
+    Array(options[:requires_gem]).all? do |g|
+      Pry::Rubygem.installed?(g)
+    end
+  end
 
-      def rbx?
-        RbConfig::CONFIG['ruby_install_name'] == 'rbx'
-      end
+  def use_ansi_codes?
+    windows_ansi? || ENV['TERM'] && ENV['TERM'] != "dumb"
+  end
 
-      def mri?
-        RbConfig::CONFIG['ruby_install_name'] == 'ruby'
-      end
+  def colorize_code(code)
+    CodeRay.scan(code, :ruby).term
+  end
 
-      def mri_19?
-        mri? && RUBY_VERSION =~ /^1\.9/
-      end
+  def highlight(string, regexp, highlight_color=:bright_yellow)
+    string.gsub(regexp) { |match| "<#{highlight_color}>#{match}</#{highlight_color}>" }
+  end
 
-      def mri_2?
-        mri? && RUBY_VERSION =~ /^2/
-      end
+  # formatting
+  def heading(text)
+    text = "#{text}\n--"
+    "\e[1m#{text}\e[0m"
+  end
 
-      # Send the given text through the best available pager (if Pry.config.pager is
-      # enabled). Infers where to send the output if used as a mixin.
-      # DEPRECATED.
-      def stagger_output(text, out = nil)
-        if defined?(_pry_) && _pry_
-          _pry_.pager.page text
-        else
-          Pry.new.pager.page text
-        end
-      end
+  # Send the given text through the best available pager (if Pry.config.pager is
+  # enabled). Infers where to send the output if used as a mixin.
+  # DEPRECATED.
+  def stagger_output(text, out = nil)
+    if defined?(_pry_) && _pry_
```

Comment:
in what situatio would `_pry_` not be defined? just curious

Reply:
singleton method: Pry::Helpers::BaseHelpers.stagger_output

Reply:
https://github.com/pry/pry/pull/1661

Reply:
https://github.com/pry/pry/issues/1660